fv3core/stencils/remapping_part2.py

```diff
@@ -27,125 +25,150 @@ def init_phis(hs: FloatField, delz: FloatField, phis: FloatField, te_2d: FloatFi
             phis = phis[0, 0, 1] - constants.GRAV * delz
 
 
-@gtstencil
 def sum_te(te: FloatField, te0_2d: FloatField):
     with computation(FORWARD):
         with interval(0, None):
             te0_2d = te0_2d[0, 0, -1] + te
 
 
-def compute(
-    qvapor: FloatField,
-    qliquid: FloatField,
-    qice: FloatField,
-    qrain: FloatField,
-    qsnow: FloatField,
-    qgraupel: FloatField,
-    qcld: FloatField,
-    pt: FloatField,
-    delp: FloatField,
-    delz: FloatField,
-    peln: FloatField,
-    u: FloatField,
-    v: FloatField,
-    w: FloatField,
-    ua: FloatField,
-    cappa: FloatField,
-    q_con: FloatField,
-    gz: FloatField,
-    pkz: FloatField,
-    pk: FloatField,
-    pe: FloatField,
-    hs: FloatFieldIJ,
-    te0_2d: FloatFieldIJ,
-    te: FloatField,
-    cvm: FloatField,
-    pfull: FloatFieldK,
-    ptop: float,
-    akap: float,
-    r_vir: float,
-    last_step: bool,
-    pdt: float,
-    mdt: float,
-    consv: float,
-    do_adiabatic_init: bool,
-):
-    saturation_adjustment = utils.cached_stencil_class(SatAdjust3d)(
-        cache_key="satadjust3d"
-    )
-
-    grid = spec.grid
-    copy_from_below(
-        ua, pe, origin=grid.compute_origin(), domain=grid.domain_shape_compute()
-    )
-    dtmp = 0.0
-    phis = utils.make_storage_from_shape(
-        pt.shape, grid.compute_origin(), cache_key="remapping_part2_phis"
-    )
-    te_2d = utils.make_storage_from_shape(
-        pt.shape[0:2], grid.compute_origin(), cache_key="remapping_part2_te_2d"
-    )
-    zsum1 = utils.make_storage_from_shape(
-        pt.shape[0:2], grid.compute_origin(), cache_key="remapping_part2_zsum1"
-    )
-    if spec.namelist.do_sat_adj:
-        fast_mp_consv = not do_adiabatic_init and consv > constants.CONSV_MIN
+class Remapping_Part2:
```

Comment:
Oliver named part 1 VerticalRemapping1, so probably makes sense to adopt his convention.